src/qt/overviewpage.cpp

```diff
@@ -181,6 +181,8 @@ void OverviewPage::setPrivacy(bool privacy)
 
     ui->listTransactions->setVisible(!m_privacy);
 
+    walletModel->getTransactionTableModel()->updateDisplayUnit();
```

Comment:
I'm not really against this line but it is a bit of a hack to trigger the column amount re-paint.

To make the intent clearer, could decouple the `dataChanged` signal into its own method, so it can be called from `updateDisplayUnit` internally and here too.

Reply:
Thanks @furszy, I took your advice on the decoupling the `dataChanged` signal and I've refactored the `updateDisplayUnit` `Q_SLOT` accordingly. As I was still thinking as you said, the call to `updateDisplayUnit` it's kind of a hack, so I've managed to add the connect on the `WalletView constructor` to trigger the `refreshAmounts` on the `setPrivacy`.